tsdb/engine/tsm1/cache.go

```diff
@@ -461,13 +469,13 @@ func (c *Cache) updateMemSize(b int64) {
 }
 
 // Update the snapshotsCount and the diskSize levels
-func (c *Cache) updateSnapshot() {
+func (c *Cache) updateSnapshots() {
```

Comment:
@jwilder I tried to explain what the purpose of this counter is. It is basically counting difference between number of calls to Cache.Snapshot() and the number of calls to ClearSnapshot().

A value of 0 means no goroutine is executing within WriteSnaphot.

A value of 1 means exactly one goroutine is executing within WriteSnapshot but has not yet called ClearSnapshot.

A value of 2 means at least two routines have entered WriteSnapshot and called Snapshot() but either are still executing within WriteSnapshot() or at least one of them exited WriteSnapshot() early without calling ClearSnapshot() (which resets that value to 0)

As you can see, a value greater than 1 has real diagnostic value - indicates the number of WriteSnapshots that have started but not full completed since the last successful WriteSnapshot.

This is why the value needs a plural name, why it needs to be incremented once on each call to Snapshots() and reset to zero on the next successful call to ClearSnapshot() indicating that the last snapshot (and all preceding it since the last successful snapshot) has been written completely to disk.

I understand that snapshot is either nil or a single value and no longer a slice. The intent of the statistic was not to measure the length of the slice. It just so happened that the length of the slice happened to be the number of commenced but not yet completed snapshots. 

Now that there is only a single snapshot and no slice, we need to use a counter for the purpose that the slice length was previously a useful shorthand for.